artichoke-backend/src/extn/core/integer/mod.rs

```diff
@@ -88,6 +89,36 @@ impl Integer {
         }
     }
 
+    unsafe extern "C" fn bit(mrb: *mut sys::mrb_state, slf: sys::mrb_value) -> sys::mrb_value {
+        fn method(interp: &Artichoke, value: Value, other: Value) -> Result<Value, ArtichokeError> {
+            let value = value.try_into::<Int>()?;
+            let other = u32::try_from(other.try_into::<Int>()?);
```

Comment:
If this code calls

```rust
other.implicitly_convert_to_int()? // defined in `Value`
```

it will properly handle this case:

```ruby
class A
  def to_int; 1; end
end

0b1011[A.new]
# => 1
```

Reply:
This and the explicit `Fatal` above means it most likely makes sense to return `Exception` as the error type from this function.